src/metaxy/metadata_store/delta.py

```diff
@@ -0,0 +1,336 @@
+            for batch_df in polars_lazy.collect().iter_slices(streaming_chunk_size):
+                deltalake.write_deltalake(
+                    table_uri,
+                    batch_df,
+                    **write_kwargs,  # type: ignore[arg-type]
+                )
+        else:
+            # Regular write: collect and write entire frame
+            if isinstance(df_polars, nw.LazyFrame):
+                df_native = df_polars.collect().to_native()
+            else:
+                df_native = df_polars.to_native()
+
+            deltalake.write_deltalake(
+                table_uri,
+                df_native,
+                **write_kwargs,  # type: ignore[arg-type]
+            )
+
+    def _drop_feature_metadata_impl(self, feature_key: FeatureKey) -> None:
+        """Drop Delta table for the specified feature using soft delete.
+
+        Uses Delta's delete operation which marks rows as deleted in the transaction log
+        rather than physically removing files.
+        """
+        import deltalake  # pyright: ignore[reportMissingImports]
+
+        table_uri = self._feature_uri(feature_key)
+
+        # Check if table exists first
+        if not self._table_exists(table_uri):
+            return
+
+        # Load the Delta table
+        delta_table = deltalake.DeltaTable(
+            table_uri,
+            storage_options=self.storage_options or None,
+            without_files=True,  # Don't track files for this operation
+        )
+
+        # Use Delta's delete operation - soft delete all rows
+        # This marks rows as deleted in transaction log without physically removing files
+        delta_table.delete()
+
+    def read_metadata_in_store(
+        self,
+        feature: FeatureKey | type[BaseFeature],
+        *,
+        filters: Sequence[nw.Expr] | None = None,
+        columns: Sequence[str] | None = None,
+    ) -> nw.LazyFrame[Any] | None:
+        """Read metadata stored in Delta for a single feature using lazy evaluation."""
+        self._check_open()
+
+        feature_key = self._resolve_feature_key(feature)
+        table_uri = self._feature_uri(feature_key)
+        if not self._table_exists(table_uri):
+            return None
+
+        # Use scan_delta for lazy evaluation
+        lf = pl.scan_delta(
+            table_uri,
+            storage_options=self.storage_options or None,
+        )
+
+        # Convert to Narwhals
+        nw_lazy = nw.from_native(lf)
+
+        # Apply filters
+        if filters is not None:
+            for expr in filters:
+                nw_lazy = nw_lazy.filter(expr)
+
+        # Apply column selection
+        if columns is not None:
+            nw_lazy = nw_lazy.select(columns)
+
+        return nw_lazy
+
+    def display(self) -> str:
+        """Return human-readable representation of the store."""
+        details = [f"path={self._display_root}"]
+        if self.storage_options:
+            details.append("storage_options=***")
```

Comment:
again what's the value of this string? I don't think it does anything useful.
At least display the keys?